lib/data/dpa/dpClient.js

```diff
@@ -716,11 +744,21 @@ class DpClient {
                     return cb(null, frag);
                 });
                 return undefined;
-            });
+            }, path);
         };
 
         return getFrag(key, log.getSerializedUids(), 0, callback);
     }
+
+    getLocations(keys) {
+        const ids = keys.map(id => parseInt(id.slice(0, 6), 16).toString(2));
+        ids.forEach((id, idx) => {
+            while (ids[idx].length < 24) {
+                ids[idx] = `0${ids[idx]}`;
```

Comment:
You can replace the while with that, but maybe check performances before :)
```
ids[idx] = `${'0'.repeat(24)}${ids[idx]}`.slice(-24);
```

Or if you're sur that ```ids[idx]``` is less than 24char you can do:
```
ids[idx] = `${'0'.repeat(24 - ids[idx].length)}${ids[idx]}`;
```
Which get better performances I think.